src/modules/utils/motordrivercontrol/drivers/TMC220X/TMC220X.cpp

```diff
@@ -0,0 +1,1870 @@
+        case 2: this->ihold_irun_register_value = val; stream->printf("ihold irun config register set to %08lX\n", val); break;
+        case 3: this->slaveconf_register_value = val; stream->printf("slaveconf config register set to %08lX\n", val); break;
+        case 4: this->tpowerdown_register_value = val; stream->printf("tpowerdown register set to %08lX\n", val); break;
+        case 5: this->tpwmthrs_register_value = val; stream->printf("tpwmthrs register set to %08lX\n", val); break;
+        case 6: this->chopconf_register_value = val; stream->printf("chopconf register set to %08lX\n", val); break;
+        case 7: this->pwmconf_register_value = val; stream->printf("pwmconf register set to %08lX\n", val); break;
+
+        default:
+            stream->printf("1: gconf register\n");
+            stream->printf("2: slaveconf register\n");
+            stream->printf("3: ihold irun register\n");
+            stream->printf("4: tpowerdown register\n");
+            stream->printf("5: tpwmthrs register\n");
+            stream->printf("6: chopconf register\n");
+            stream->printf("7: pwmconf register\n");
+            stream->printf("255: update all registers\n");
+            return false;
+    }
+    return true;
+}
+
+/*
+uint8_t TMC220X::calcCRC(uint8_t *buf, uint8_t len) {
+    uint8_t crc = 0;
+	for (uint8_t i = 0; i < len; i++) {
+		uint8_t currentByte = buf[i];
+		for (uint8_t j = 0; j < 8; j++) {
+			if ((crc >> 7) ^ (currentByte & 0x01)) {
+				crc = (crc << 1) ^ 0x07;
+			} else {
+				crc = (crc << 1);
+			}
+			crc &= 0xff;
+			currentByte = currentByte >> 1;
+		}
+	}
+	return crc;
+}
+*/
+
+/*
+ * send register settings to the stepper driver via UART
+ * send 64 bits for write request and 32 bit for read request
+ * receive 64 bits only for read request
+ * returns received data content
+ */
+uint32_t TMC220X::transceive220X(uint8_t reg, uint32_t datagram)
+{
+    uint8_t rbuf[9]; // 8 + 1 crc
+    uint32_t i_datagram = 0;
+    if(reg & TMC220X_WRITE) {
+        uint8_t buf[] {(uint8_t)(TMC220X_SYNC), slave_addr, (uint8_t)(reg), (uint8_t)(datagram >> 24), (uint8_t)(datagram >> 16), (uint8_t)(datagram >> 8), (uint8_t)(datagram >> 0), (uint8_t)(0x00)};
+
+        //calculate checksum
+        calc_crc(buf, 8);
+
+        //write/read the values
+        serial(buf, 8, rbuf);
+
+        // THEKERNEL->streams->printf("sent: %02X, %02X, %02X, %02X, %02X, %02X, %02X, %02X \n", buf[0], buf[1], buf[2], buf[3], buf[4], buf[5], buf[6], buf[7]);
+    } else {
+        // reading from UART
+        uint8_t buf[] {(uint8_t)(TMC220X_SYNC), (uint8_t)(TMC220X_SLAVEADDR), (uint8_t)(reg), (uint8_t)(0x00)};
+
+        //calculate checksum
+        calc_crc(buf, 4);
+
+        //write/read the values
+        serial(buf, 4, rbuf);
+        
+        // save response CRC
+        response_crc = rbuf[5];
+        
+        calc_crc(rbuf,4);
```

Comment:
Shouldn't we be calculating crc on the whole message length of 8 bytes?  Will this put the calculated CRC in `rbuf[5]` and mess up the `i_datagram` below?